homeassistant/components/mqtt/config_flow.py

```diff
@@ -206,15 +226,164 @@ def async_get_options_flow(
         """Get the options flow for this handler."""
         return MQTTOptionsFlowHandler(config_entry)
 
+    async def _async_install_addon(self) -> None:
+        """Install the Mosquitto Mqtt broker add-on."""
+        addon_manager: AddonManager = get_addon_manager(self.hass)
+        await addon_manager.async_schedule_install_addon()
+
+    async def async_step_install_failed(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Add-on installation failed."""
+        return self.async_abort(
+            reason="addon_install_failed",
+            description_placeholders={"addon": self._addon_manager.addon_name},
+        )
+
+    async def async_step_install_addon(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Install Mosquitto Broker add-on."""
+        if self.install_task is None:
+            self.install_task = self.hass.async_create_task(self._async_install_addon())
+
+        if not self.install_task.done():
+            return self.async_show_progress(
+                step_id="install_addon",
+                progress_action="install_addon",
+                progress_task=self.install_task,
+            )
+
+        try:
+            await self.install_task
+        except AddonError as err:
+            _LOGGER.error(err)
+            return self.async_show_progress_done(next_step_id="install_failed")
+        finally:
+            self.install_task = None
+
+        return self.async_show_progress_done(next_step_id="start_addon")
+
+    async def async_step_start_failed(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Add-on start failed."""
+        return self.async_abort(
+            reason="addon_start_failed",
+            description_placeholders={"addon": self._addon_manager.addon_name},
+        )
+
+    async def async_step_start_addon(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Start Mosquitto Broker add-on."""
+        if not self.start_task:
+            self.start_task = self.hass.async_create_task(self._async_start_addon())
+        if not self.start_task.done():
+            return self.async_show_progress(
+                step_id="start_addon",
+                progress_action="start_addon",
+                progress_task=self.start_task,
+            )
+        try:
+            await self.start_task
+        except (AddonError, AbortFlow) as err:
```

Comment:
Why does the start task raise `AbortFlow` which does not abort the flow?

Reply:
Corrected. Now `AddonError` is raised instead.